WebHDFS.Kitty/WebHdfsClient.cs

```diff
@@ -184,6 +214,23 @@ public async Task Append(string path, Stream fileStream, int? buffersize = null)
             }
         }
 
+        public async Task<bool> Rename(string path, string destination)
+        {
+            var requestUri = $"/webhdfs/v1/{path.TrimStart('/')}?op=RENAME&destination=" + destination;
+
+            var request = new HttpRequestMessage(HttpMethod.Put, requestUri);
+            var response = await _httpClient.SendAsync(request);
+            if (!response.IsSuccessStatusCode)
+            {
+                var notSuccessContent = await response.Content.ReadAsStringAsync();
+                throw new InvalidOperationException($"Not success status code. Code={response.StatusCode}. Content={notSuccessContent}");
+            }
+
+            var content = await response.Content.ReadAsStringAsync();
+            var deserializedContent = JsonConvert.DeserializeObject<MakeDirectoryResponse>(content);
```

Comment:
1) Create base response class for simple boolean responses, e.g. BoolResponse
2) Create response class for every request type
3) Inherit from BoolResponse